File: ideaproject/javaserializer/testsrc/com/houseelectrics/serializer/test/path/PathTest.java
package com.houseelectrics.serializer.test.path;

import java.util.ArrayList;
import java.util.List;

import com.houseelectrics.serializer.path.PathUtil;
import org.junit.*;

/**
 * Created by roberttodd on 04/12/2014.
 */
public class PathTest
{

    public static class PathTestType
    {
        public static class Subtype
        {
            public static class MoreSub {
                private Object theItem;
                public Object getTheItem() {return theItem;}
                public void setTheItem(Object value) {this.theItem=value;}
            }
            public Subtype() { theSubSubs = new ArrayList<MoreSub>(); }
            private List<MoreSub> theSubSubs;
            public List<MoreSub> getTheSubSubs() { return theSubSubs; }
            public void setTheSubSubs(List<MoreSub> value) {this.theSubSubs=value;}
        }
        public Subtype theSub;
        public Subtype getTheSub() { return theSub; }
        public void setTheSub(Subtype value) {this.theSub = value;}
        public PathTestType() { theSub = new Subtype(); }

    }
    @Test
    public void testPath()
    {
        PathTestType testItem = new PathTestType();
        List<PathTestType.Subtype.MoreSub> subsubs = testItem.getTheSub().getTheSubSubs();
        PathTestType.Subtype.MoreSub moreSub;
        Object object2find = new Object();

        moreSub = new PathTestType.Subtype.MoreSub();
        moreSub.setTheItem(new Object());
        subsubs.add(moreSub);

        moreSub = new PathTestType.Subtype.MoreSub();
        moreSub.setTheItem(object2find);
        subsubs.add(moreSub);

        moreSub = new PathTestType.Subtype.MoreSub();
        moreSub.setTheItem( new Object());
        subsubs.add(moreSub);

        PathUtil util = new PathUtil();

        List<PathUtil.InstancePathElement> path = util.findSingleObjectPath(testItem, object2find);
        //path is TheSub.TheSubSubs.[1]
        String strPath = util.toString(path);
        System.out.println("path= " + strPath);
        Assert.assertEquals( "expected path", 4, path.size());
        String expectedPathString = "TheSub.TheSubSubs.[1].TheItem";
        Assert.assertEquals("expected path string ", expectedPathString, strPath);

    }

}

File: ideaproject/javaserializer/testsrc/com/houseelectrics/serializer/test/path/PathTest.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using NUnit.Framework;
using com.houseelectrics.serializer.path;

namespace com.houseelectrics.serializer.test.path
{
    public class PathTest
    {
        public class PathTestType
        {
            public class Subtype
            {
                public class MoreSub {
                public object TheItem {get; set;}
                }
                public Subtype() { TheSubSubs = new List<MoreSub>(); }
                public List<MoreSub> TheSubSubs { get; set; }
            }
            public Subtype TheSub { get; set; }
            public PathTestType() { TheSub = new Subtype(); }


        }
        [Test]
        public void testPath()
        {
            PathTestType testItem = new PathTestType();
            List<PathTestType.Subtype.MoreSub> subsubs = testItem.TheSub.TheSubSubs;
            PathTestType.Subtype.MoreSub moreSub;
            object object2find = new object();

            moreSub = new PathTestType.Subtype.MoreSub();
            moreSub.TheItem = new object();
            subsubs.Add(moreSub);

            moreSub = new PathTestType.Subtype.MoreSub();
            moreSub.TheItem = object2find;
            subsubs.Add(moreSub);

            moreSub = new PathTestType.Subtype.MoreSub();
            moreSub.TheItem = new object();
            subsubs.Add(moreSub);

            PathUtil util = new PathUtil();

            List<InstancePathElement> path = util.findSingleObjectPath(testItem, object2find);
            //path is TheSub.TheSubSubs.[1]
            string strPath = util.ToString(path);
            System.Console.WriteLine("path= " + strPath);
            Assert.AreEqual(4, path.Count, "expected path");
            string expectedPathString = "TheSub.TheSubSubs.[1].TheItem";
            Assert.AreEqual(expectedPathString, strPath, "expected path string ");

        }
    }
}
